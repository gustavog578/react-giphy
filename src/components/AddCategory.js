import React, { useState } from 'react'
import { PropTypes } from "prop-types";

export const AddCategory = ({setCategories}) => {
// poner '' para que el state inicial no sea undefined
const [Inputvalue, setInputvalue] = useState('')



const handleInputChange = (e) => {
    setInputvalue(e.target.value);
}


const handleSubmit = (e) => {
    e.preventDefault();
    
    if(Inputvalue.trim().length > 2){
        setCategories(cats => [Inputvalue, ...cats])
        setInputvalue('')
    }

}


return (
   
        <form onSubmit={ handleSubmit }>         
            <input
                type="text"
                value={ Inputvalue }
                onChange={ handleInputChange }
            />
        </form>


    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}