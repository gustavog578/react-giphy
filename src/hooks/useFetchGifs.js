
import  { useEffect, useState } from 'react'
import { getGifs } from "../helpers/getGifs";

// los effect no pueden ser async 

export const useFetchGifs = ( category ) => {
    
    const [state, setState] = useState({
        data : [],
        loading: true
    })
    // el componente solo debe renderizarse y cargarse cuando cambia la categoria
    useEffect( ()=> {
            getGifs( category)
            .then( imgs => {
                    setState({
                        data: imgs,
                        loading:false
                    })                   
            } )
    },[category])

    return state;

}

