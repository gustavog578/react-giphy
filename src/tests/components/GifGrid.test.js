import React from 'react';
import '@testing-library/jest-dom';

import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas sobre Gif Grid', ()=> {

    const category = 'One Punch';


    test('Debe mostrarse correctamente', ()=> {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GifGrid category={category}/>)   
        expect(wrapper).toMatchSnapshot()

    });

    test('Debe mostrat items cuando se cargan imagenes en useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/algo/algo',
            title: 'Algo'
        },{
            id: 'ABC1',
            url: 'http://localhost/algo/algo',
            title: 'Algo'
        }] 

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        })

        const wrapper = shallow( <GifGrid category={category}/>);

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists()).toBe(false)
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length)

    })




})