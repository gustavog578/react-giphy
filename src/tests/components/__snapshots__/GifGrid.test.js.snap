// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre Gif Grid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre Gif Grid Debe mostrat items cuando se cargan imagenes en useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Algo"
      url="http://localhost/algo/algo"
    />
    <GifGridItem
      id="ABC1"
      key="ABC1"
      title="Algo"
      url="http://localhost/algo/algo"
    />
  </div>
</Fragment>
`;
